qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/SemanticSearchIT.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.bwc;
+
+import com.carrotsearch.randomizedtesting.RandomizedTest;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static org.opensearch.neuralsearch.TestUtils.NODES_BWC_CLUSTER;
+
+import static org.opensearch.neuralsearch.TestUtils.getModelId;
+import static org.opensearch.neuralsearch.TestUtils.TEXT_EMBEDDING_PROCESSOR;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+public class SemanticSearchIT extends AbstractRestartUpgradeRestTestCase {
+
+    private static final String PIPELINE_NAME = "nlp-pipeline";
+    private static String DOC_ID = "0";
+    private static final String TEST_FIELD = "passage_text";
+    private static final String TEXT = "Hello world";
+
+    // Test restart-upgrade Semantic Search
+    // Create Text Embedding Processor, Ingestion Pipeline and add document
+    // Validate process , pipeline and document count in restart-upgrade scenario
+    public void testSemanticSearch() throws Exception {
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+
+        if (isRunningAgainstOldCluster()) {
+            String modelId = uploadTextEmbeddingModel();
+            loadModel(modelId);
+            createPipelineProcessor(modelId, PIPELINE_NAME);
+            createIndexWithConfiguration(
+                testIndex,
+                Files.readString(Path.of(classLoader.getResource("processor/IndexMappings.json").toURI())),
+                PIPELINE_NAME
+            );
+            addDocument(testIndex, DOC_ID, TEST_FIELD, TEXT);
+        } else {
+            Map<String, Object> pipeline = getIngestionPipeline(PIPELINE_NAME);
+            assertNotNull(pipeline);
+            String modelId = getModelId(pipeline, TEXT_EMBEDDING_PROCESSOR);
+            validateTestIndex(modelId);
```

Comment:
Should we also test that we can add another document to the index - this would ensure the processor created on the old version works on the new version.

Reply:
Yes I can